packages/fe/store/dataset.js

```diff
@@ -119,19 +51,42 @@ const actions = {
     }
   },
   // //////////////////////////////////////////////////////////////// getCidList
-  getCidList ({ commit, getters }, metadata) {
+  async getCidList ({ commit, getters, dispatch }, metadata) {
     try {
-      // const id = metadata.dataSetId
-      // const response = await this.$axiosAuth('/get-cid-list',  { params: { id } })
-      // const response = await getMockCids()
-      // const cids = response.data.payload
-      const cids = false
-      commit('SET_CID_LIST', { cids })
+      dispatch('setCidLoadingStatus', { status: true })
+      const route = metadata.route
+      const query = CloneDeep(route.query)
+      const slug = route.params.id
+      const page = parseInt(query.page || getters.metadata.page)
+      const limit = query.limit || getters.metadata.limit
+      const search = query.search
+      const response = await this.$axiosAuth('/get-cid-list', {
+        params: {
+          slug,
+          page,
+          ...(limit && { limit }),
+          ...(search && { search })
+        }
+      })
+      const payload = response.data.payload
+      dispatch('setCidList', {
+        results: payload.results,
+        metadata: payload.metadata
+      })
```

Comment:
Need to run `dispatch('setCidLoadingStatus', { status: false })` at the end of the `try` block as well (not just in the `catch` block). Otherwise it looks like it's loading forever *tear

<img width="1333" alt="Screenshot 2023-03-07 at 1 25 20 PM" src="https://user-images.githubusercontent.com/11708190/223531906-d054904d-ed31-4352-8a41-fb736d5badaa.png">

Reply:
@timelytree the `setCidLoadingStatus` action is called and set to false in the CID table actually... there's a watcher watching the cidList in the store - whenever it's changed the watcher will trigger a call to set the loading status to false. That should cover all cases.

Reply:
Ah that's great actually 🙂 althoughhh I'm getting an infinite loader when landing on a singular page 🤔

Reply:
@timelytree which one? was it one without any CIDs? or did the spinner stay even after CIDs had loaded?

Reply:
Resolved! 🙂